File: LMIS/LMIS.Web/Modules/ReaderManage/LibraryCard/LibraryCardColumns.cs
using LMIS.Modules.ReaderManage.LibraryCard;
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace LMIS.ReaderManage.Columns;

[ColumnsScript("ReaderManage.LibraryCard")]
[BasedOnRow(typeof(LibraryCardRow), CheckNames = true)]
public class LibraryCardColumns
{
    [EditLink,AlignRight]
    public long LibraryCardId { get; set; }
    [EditLink]
    public string LibraryCardNo { get; set; }
    public string UserName { get; set; }
    public string CardLevelName { get; set; }
    public LibraryCardStatusEnum Status { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime CreateTime { get; set; }
    [Width(150)]
    [DateTimeFormatter(DisplayFormat = "yyyy/MM/dd HH:mm:ss")]
    public DateTime UpdateTime { get; set; }
}

File: LMIS/LMIS.Web/Modules/ServerTypes/ReaderManage/LibraryCardColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { LibraryCardStatusEnum } from "../Modules/ReaderManage.LibraryCard.LibraryCardStatusEnum";
import { LibraryCardRow } from "./LibraryCardRow";

export interface LibraryCardColumns {
    LibraryCardNo: Column<LibraryCardRow>;
    UserName: Column<LibraryCardRow>;
    CardLevelName: Column<LibraryCardRow>;
    Status: Column<LibraryCardRow>;
    CreateTime: Column<LibraryCardRow>;
    UpdateTime: Column<LibraryCardRow>;
}

export class LibraryCardColumns extends ColumnsBase<LibraryCardRow> {
    static readonly columnsKey = 'ReaderManage.LibraryCard';
    static readonly Fields = fieldsProxy<LibraryCardColumns>();
}

[LibraryCardStatusEnum]; // referenced types